qa/tasks/cephfs/test_cephfs_shell.py

```diff
@@ -83,8 +84,16 @@ def run_cephfs_shell_script(self, script, mount_x=None,
         scriptpath = tempfile_mkstemp(prefix='test-cephfs', text=True)[1]
         with open(scriptpath, 'w') as scriptfile:
             scriptfile.write(script)
-        # copy script to the machine running cephfs-shell.
-        mount_x.client_remote.put_file(scriptpath, scriptpath)
+
+        # copy script to the machine running cephfs-shell. in case of
+        # vstart_runner.py, these this shutil.SameFileError would be raised
+        # since source and destination path would be the same, let's expect
+        # and ignore that.
```

Comment:
@batrick I've added the comment as we discussed during our last meeting. If no more changes are required, can we please merge this?

Reply:
I see now. Since this problem is specific to `vstart_runner.py`, wouldn't it be better to put the exception catch in the `put_file` hook in `vstart_runner.py`? In fact, it looks like the code is already trying to do this:

https://github.com/ceph/ceph/blob/61fcf90d8a8bc029ded400321f04a3190d7eb3b1/qa/tasks/vstart_runner.py#L274-L277

Reply:
That code was added by me IIRC... Since, I couldn't encounter the exception with python2 I thought this was due to upgrade in python3. Now, I see the real reason.

I've made the change. Thanks!